app/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java

```diff
@@ -289,21 +293,61 @@ public void setEncryptionAttributeForItemID(String fileId, boolean encrypted) {
         new Handler(Looper.getMainLooper()).post(this::notifyDataSetChanged);
     }
 
-
     @Override
     public long getItemId(int position) {
-        if (mFiles == null || mFiles.size() <= position) {
-            return 0;
+        boolean isFilesEmpty = (mFiles == null || mFiles.size() <= position);
+
+        if (isFilesEmpty) {
+            if (shouldShowHeader()) {
+                return headerId;
+            } else {
+                return footerId;
+            }
+        } else {
+            return mFiles.get(position).getFileId();
         }
```

Comment:
```suggestion
        if (shouldShowHeader()) {
            if (position == 0) {
                return headerId;
            }

            // skip header
            position--;
        }

        if (position == mFiles.size()) {
            return footerId;
        } if (position < mFiles.size()) {
            return mFiles.get(position).getFileId();
        }

        // fallback
        return RecyclerView.NO_ID;
```

With this patch in place, it no longer crashes on my test device.